sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -972,6 +1090,540 @@ public async Task StopProcessingLogsFaultedTaskDuringShutdown(bool async)
+                .Verify("OnProcessingErrorAsync", Times.Once(),
+                     expectedException,
+                     ItExpr.IsNull<EventProcessorPartition>(),
+                     Resources.OperationGetPartitionIds,
+                     CancellationToken.None);
+
+            if (async)
+            {
+                mockProcessor
+                    .Verify(processor => processor.StopProcessingAsync(
+                        CancellationToken.None),
+                    Times.Once);
+            }
+            else
+            {
+                mockProcessor
+                    .Verify(processor => processor.StopProcessing(
+                        CancellationToken.None),
+                    Times.Once);
+            }
+
+            // Due to the required ordering of the mocking callback when delegating to the base implementation, there is a race condition in
+            // which the processor may be stopping when the completion source is set; wait a small bit to allow the state to finalize.
+
+            for (var attempt = 0; attempt < 10; ++attempt)
+            {
+                if (mockProcessor.Object.Status != EventProcessorStatus.Stopping)
+                {
+                    break;
+                }
+
+                await Task.Delay((attempt + 1) * 50);
+            }
+
+            Assert.That(mockProcessor.Object.Status, Is.EqualTo(EventProcessorStatus.NotRunning), "The processor should not be running.");
+            Assert.That(capturedException, Is.Null, "No exception should have occurred when stopping the processor.");
+
+            cancellationSource.Cancel();
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   background processing loop.
+        /// </summary>
+        ///
+        [Test]
+        public async Task BackgroundProcessingDispatchesTopLevelExceptions()
+        {
+            using var cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(15));
+
+            var expectedException = new DivideByZeroException("BOOM!");
+            var completionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            var mockProcessor = new Mock<EventProcessor<EventProcessorPartition>>(65, "consumerGroup", "namespace", "eventHub", Mock.Of<TokenCredential>(), default(EventProcessorOptions)) { CallBase = true };
+
+            mockProcessor
+                .Setup(processor => processor.CreateConnection())
+                .Throws(expectedException);
+
+            mockProcessor
+               .Protected()
+               .Setup<Task>("OnProcessingErrorAsync", expectedException, ItExpr.IsAny<EventProcessorPartition>(), ItExpr.IsAny<string>(), ItExpr.IsAny<CancellationToken>())
+               .Callback(() => completionSource.TrySetResult(true))
+               .Returns(Task.Delay(TimeSpan.FromSeconds(20)));
```

Comment:
These are intentionally a longer duration than the cancellation token timeout; my intent is to prove for each of these scenarios that the error handler does not block other operations.   I considered covering that in an additional test, but there are a non-trivial number of cases and this isn't a vital core scenario that I believe requires a specific dedicated test.

Reply:
You're right, of course, and I was being lazy because it cascades into a few other tests.  I've been shamed into doing the right thing.   😄